### ISSUE TITLE
Try Catch not working if exception thrown in overriden abstract method

### ISSUE THREAD

**Issue by retrievaaaa** [CosmosOS/Cosmos#2313]
#### Area of Cosmos - What area of Cosmos are we dealing with?
Runtime

#### Expected Behaviour - What do you think that should happen?
The catch block should be called in case of an exception

#### Actual Behaviour - What unexpectedly happens?
The OS just crashes

#### Reproduction - How did you get this error to appear?
So for example, take this Run Method:
```
protected override void Run() {
    try {
        ProcessManager.StartProcess(new Process(new TaskbarApp()));

        while (true) {
            foreach (Process proc in ProcessManager.GetProcesses()) {
                proc.app.Update();
            }

            graphics.RenderFrame();
            frameCounter++;

            if(previousSecond != RTC.Second) {
                previousSecond = RTC.Second;
                FramesPerSecond = frameCounter;
                frameCounter = 0;
            }
        }
    }catch(Exception ex) {
        mDebugger.SendMessage("exception", ex.Message);
        Panic.KernelPanic("GENERIC_RunningKernel", ex);
    }
}
```

Now imagine, that StartProcess adds the process to a list. Now, if the list is uninitiliazed, the OS will crash and the catch block will not be executed. It might be worth noting that ProcessManager is a static class, however that shouldn't really matter.

#### Version - Were you using the User Kit or Dev Kit? And what User Kit version or Dev Kit commit (Cosmos, IL2CPU, X#)?
DevKit Commit https://github.com/CosmosOS/Cosmos/commit/d64e68add45ef95c4163e79407fe8f762b7bb5f3

**Comment by AsertCreator** [comment]
when i use try catch, it sometimes works sometimes it crashes cosmos, not os.

**Comment by GoldenretriverYT** [comment]
Yeah it does work SOMETIMES for me too, but sometimes the OS then just freezes or crashes

**Comment by 9xbt** [comment]
that's weird, it never does that to me
it actually calls the BSOD function i have so yeah, idk what happened

**Comment by audiotore482** [comment]
Not sure if this is too late but I believe on the (catch) line you have to put a "Console.ReadKey();"
Thats because when something goes wrong cosmos resorts to catch and executes all the code
in their and after that Cosmos Doesn't know what to do so it now resorts to crashing. If you put a 
Console.ReadKey(); it will give you time to read the error message and see whats going On.

**Comment by GoldenretriverYT** [comment]
> Not sure if this is too late but I believe on the (catch) line you have to put a "Console.ReadKey();" Thats because when something goes wrong cosmos resorts to catch and executes all the code in their and after that Cosmos Doesn't know what to do so it now resorts to crashing. If you put a Console.ReadKey(); it will give you time to read the error message and see whats going On.

Has absolutely nothing to do with this, it does not execute the code at all. My kernel panic screen already halts with a loop. And debugger messages are also not sent, which means that it never executes this.

**Comment by audiotore482** [comment]
Can you just try printing out the ex variable then?

**Comment by GoldenretriverYT** [comment]
What do you mean? Where?
It is not executed. Thats a fact. Nothing will change that. I could literally change it to send "a" to the debugger and it wont execute.

**Comment by GoldenretriverYT** [comment]
What I just noticed, this seems to only happen if your error occurs within an overriden abstract method. Apart from that, try/catch works perfectly fine.

**Comment by valentinbreiz** [comment]
Cosmos Gen3 has replaced the IL2CPU-based Gen2 codebase, and `gen3` is now the default branch of this repository. This issue was filed against Gen2, which is no longer developed, so it is being closed as part of the Gen3 cutover (#3088).

If you still hit this on Gen3, please open a fresh issue. The compiler, the build pipeline and the debugging story are all different now, so a new report will get a far more useful answer than reviving this one. Start here: https://cosmosos.github.io/

Gen2 remains on the `gen2` branch, and this issue stays readable.
